open_spiel/games/phantom_go.cc

```diff
@@ -0,0 +1,793 @@
+
+    if(state1.board_.board_size() != state2.board_.board_size())
+    {
+        return false;
+    }
+
+    std::array<int, 2> stoneCount1 = state1.board_.GetStoneCount();
+    std::array<int, 2> stoneCount2 = state2.board_.GetStoneCount();
+
+    if(stoneCount1[0] != stoneCount2[0] || stoneCount1[1] != stoneCount2[1])
+    {
+        return false;
+    }
+
+    int boardSize = state1.board_.board_size();
+
+    auto observation1 = state1.board_.GetObservationByID(playerID);
+    auto observation2 = state2.board_.GetObservationByID(playerID);
+
+    for(int i = 0; i < boardSize * boardSize; i++)
+    {
+        if(observation1[i] != observation2[i])
+        {
+            return false;
+        }
+    }
+
+    if(state1.to_play_ != state2.to_play_)
+    {
+        return false;
+    }
+
+    return true;
+}
+
+PhantomGoGame::PhantomGoGame(const GameParameters &params)
+    : Game(kGameType, params),
+      komi_(ParameterValue<double>("komi")),
+      board_size_(ParameterValue<int>("board_size")),
+      handicap_(ParameterValue<int>("handicap")),
+      max_game_length_(ParameterValue<int>(
+          "max_game_length", DefaultMaxGameLength(board_size_))) {}
+
+class PhantomGoObserver : public Observer {
+ public:
+  PhantomGoObserver(IIGObservationType iig_obs_type)
+      : Observer(/*has_string=*/true, /*has_tensor=*/true),
+        iig_obs_type_(iig_obs_type) {}
+
+  void WriteTensor(const State &observed_state, int player,
+                   Allocator *allocator) const override {
+      const PhantomGoState &state =
+          open_spiel::down_cast<const PhantomGoState &>(observed_state);
+
+      const int totalBoardPoints = state.board().board_size() * state.board().board_size();
+
+      {
+          auto out = allocator->Get("stone-counts", {2});
+          auto stoneCount = state.GetStoneCount();
+          out.at(0) = stoneCount[0];
+          out.at(1) = stoneCount[1];
+      }
+
+      if (iig_obs_type_.private_info == PrivateInfoType::kSinglePlayer) {
+          {
+              auto out = allocator->Get("player_observation", {totalBoardPoints});
+              auto observation = state.board().GetObservationByID(player);
+              for (int i = 0; i < totalBoardPoints; i++) {
+                  out.at(i) = (uint8_t) observation[i];
+              }
+          }
+      }
+
+      if (iig_obs_type_.public_info) {
+
+          {
+              auto out = allocator->Get("history-turns", {state.History().size()});
+              auto history = state.FullHistory();
+              for (int i = 0; i < history.size(); i++) {
+                  out.at(i) = history[i].player;
+              }
+          }
+
+          {
+              std::shared_ptr<const Game> game = state.GetGame();
+              std::unique_ptr<PhantomGoState>
+                  currState = std::make_unique<PhantomGoState>(down_cast<PhantomGoState>(*game->NewInitialState()));
+              auto out = allocator->Get("history-turns", {state.History().size()});
```

Comment:
Choose a different field name. `history-turns` is already used.

Reply:
I believe this should encode the stone count differences between turns? Can you encode only the current difference?